ironfish-rust/src/merkle_note.rs

```diff
@@ -93,9 +89,51 @@ impl MerkleNote {
             &note.commitment_point(),
             public_key,
         );
-        let mut note_encryption_keys = [0; ENCRYPTED_SHARED_KEY_SIZE + aead::MAC_SIZE];
+        let mut note_encryption_keys = [0; NOTE_ENCRYPTION_KEY_SIZE];
         aead::encrypt(&encryption_key, &key_bytes, &mut note_encryption_keys);
 
+        Self::construct(
+            note,
+            value_commitment,
+            diffie_hellman_keys,
+            note_encryption_keys,
+        )
+    }
+
+    /// Helper function to instantiate a MerkleNote with pre-set
+    /// note_encryption_keys. Should only be used for miners fee transactions.
+    pub(crate) fn new_for_miners_fee(
```

Comment:
This whole code path feels a bit crufty, but was necessary to set the hard-coded encryption keys before the build step is executed. I think eventually, I'd like to create wrappers specifically for miners fee so we can separate edge-cased logic that exists specifically for miners fees from regular code paths.